packages/fluentui/docs/src/examples/components/Button/Visual/ButtonExampleCompose.shorthand.tsx

```diff
@@ -0,0 +1,163 @@
+import { compose, Button, ButtonProps, ButtonStylesProps, Flex, Header, Provider } from '@fluentui/react-northstar';
+import { ComponentSlotStylesInput, ComponentVariablesInput, ThemeInput } from '@fluentui/styles';
+import * as React from 'react';
+
+//
+// Components
+//
+
+// Adds a custom design term
+//
+
+type TertiaryButtonProps = {
+  tertiary?: boolean;
+};
+
+type TertiaryButtonStylesProps = TertiaryButtonProps;
+
+const ComposedButton = compose<TertiaryButtonProps, TertiaryButtonStylesProps, ButtonProps, {}>(Button, {
+  className: 'ui-tertiary-button',
+  displayName: 'TertiaryButton',
+  mapPropsToStylesProps: props => ({ tertiary: props.tertiary }),
+  handledProps: ['tertiary'],
+});
+
+// Adds overrides for a design term
+//
+
+const CompactTertiaryButton = compose(ComposedButton, {
+  displayName: 'CompactTertiaryButton',
+});
+
+// Composes custom button
+//
+
+type OverriddenButtonProps = {
+  fitted?: boolean;
+};
+
+type OverriddenButtonStylesProps = Required<OverriddenButtonProps>;
+
+const OverriddenButton = compose<OverriddenButtonProps, OverriddenButtonStylesProps, ButtonProps, {}>(Button, {
+  className: 'ui-overridden-button',
+  displayName: 'OverriddenButton',
+  mapPropsToStylesProps: props => ({
+    fitted: props.fitted,
```

Comment:
It would be good if we can use also some other style prop here from the `Button`, just as a showcase that we have access to them too. I would propose `primary` or `disabled`.

Reply:
![image](https://user-images.githubusercontent.com/14183168/77433969-6999a400-6de0-11ea-8186-588fc4a0052c.png)

Agree, added 👍